core/frontend/src/RealityDataFormatRegistry.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Tiles
+ */
+
+/** Returns the URL used for retrieving Google Photorealistic 3D Tiles.
+ * @alpha
+ */
+export function getGooglePhotorealistic3DTilesURL() {
+  return "https://tile.googleapis.com/v1/3dtiles/root.json";
+}
+
+/** Stores key-value pair to be added to reality data requests.
+ * @alpha
+ */
+export interface RealityDataKey {
+  key: string;
+  value: string;
+}
+
+/** Options supplied at startup via [[IModelAppOptions.realityDataOptions]] to specify access keys for various reality data formats.
+ * `gp3dt` must have its key value set to `key`
+ * @alpha
+ */
+export interface RealityDataOptions {
+  /** Access key for Google Photorealistic 3D Tiles in the format `{ key: "key", value: "your-gp3dt-key" }`. */
+  gp3dt?: RealityDataKey;
+}
+
+/**
+ * A registry of RealityDataFormats identified by their unique format IDs. The registry can be accessed via [[IModelApp.realityDataFormatRegistry]].
+ * @alpha
+ */
+export class RealityDataFormatRegistry {
```

Comment:
This should probably be a *provider* registry not *format* - google's tiles are in the same format as ION's, Context Capture's, etc - what differs is the provider logic (authentication, url formatting, etc). That would match with `RealityDataSourceKey.provider`.

Reply:
My takeaways from this feedback:
1) Make this be `RealityDataProviderRegistry`.
2) Add something [like this](https://github.com/iTwin/itwinjs-core/blob/c9958802ae321dbf54dd7f4eea74c429fad6da51/core/frontend/src/tile/map/MapLayerFormatRegistry.ts#L113) to `RealityDataOptions`.

Is there something further you are thinking of?  It does just basically wrap an auth key right now.

Reply:
No, the crucial thing your registry is missing compared to `MapLayerFormatRegistry` is the `register` function. The "config" thing with key-value pairs IMO is not great; it'd make more sense to me if somebody instantiates a `Google3DTilesProvider`, passing it the API key and whatever else it needs, and then registers that with your registry.